data/collector_test.go

```diff
@@ -264,12 +264,15 @@ func TestCtxCancelledNotLoggedAfterClose(t *testing.T) {
 
 	params := CollectorParams{
 		ComponentName: "testComponent",
-		Interval:      time.Millisecond * 1,
-		MethodParams:  map[string]*anypb.Any{"name": fakeVal},
-		Target:        target,
-		QueueSize:     queueSize,
-		BufferSize:    bufferSize,
-		Logger:        logger,
+		// Ensure that we use ticker-based capture since capturing at 1000+ Hz
+		// uses a sleep-based capture that may let a context cancelation error
+		// occasionally be logged.
+		Interval:     sleepCaptureCutoff + time.Microsecond,
```

Comment:
sleepCaptureCutoff is [2ms](https://github.com/viamrobotics/rdk/blob/a81e17bcc2ec78c729018e7d8f7d3c445d9ab665/data/collector.go#L30-L31). We can tweak this later to be closer to 1ms if we get more users capturing at high frequencies.